### PR TITLE
PE-795: BATS for file and drive name validation

### PR DESCRIPTION
**Pull request by karlprieb**

### CODE REVIEW

**Review by fedellen — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -249,6 +249,7 @@ To ensure your environment is compatible, we also recommend the following VSCode
 
 **Number of files in a bulk upload:** Theoretically unlimited<br>
 **Max individual file size**: 2GB (Node.js limitation)<br>
+**Max file name length**: 254 characters<br>
```
Comment (arielmelendez) [inline README.md:252]: Isn't the limit 255 characters? That's what we agreed on in the ArFS Standards writeup...

File: bats_test/create-drive/drive-name-validation.bats [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+load '/home/node/packages/node_modules/bats-assert/load.bash'
+
+@test "create drive with empty name" {
+    run -1 bash -c "yarn ardrive create-drive --dry-run -n '' -w $WALLET"
+
+    assert_line -n 0 "Error: Required parameter driveName wasn't provided!"
+}
+
+@test "create drive with long name" {
+    run -1 bash -c "yarn ardrive create-drive --dry-run -n '+===============================================================================================================================================================================================================================================================' -w $WALLET"
```
Comment (arielmelendez) [inline bats_test/create-drive/drive-name-validation.bats:15]: Might be nice to factor out this long name as a string, assert that its length is 255 characters before the test starts and then use the variable in the ardrive invocation.

File: bats_test/create-drive/drive-name-validation.bats [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bats
+
+# Support lib
+load '/home/node/packages/node_modules/bats-support/load.bash'
+# Assertions
+load '/home/node/packages/node_modules/bats-assert/load.bash'
+
+@test "create drive with empty name" {
```
Comment (arielmelendez) [inline bats_test/create-drive/drive-name-validation.bats:8]: These test names should not just be what the test inputs are. It should also include the expected behavior, e.g.
```
@test "create drive with empty name results in an error message and exit code 1" {
```

File: bats_test/create-drive/drive-name-validation.bats [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    run -1 bash -c "yarn ardrive create-drive --dry-run -n '' -w $WALLET"
+
+    assert_line -n 0 "Error: Required parameter driveName wasn't provided!"
```
Comment (arielmelendez) [inline bats_test/create-drive/drive-name-validation.bats:11]: You should also test for the exit status code (if possible)

Comment (karlprieb) [reply bats_test/create-drive/drive-name-validation.bats:11]: It's testing for exit status 1 on `run -1`

File: bats_test/create-drive/drive-name-validation.bats (hunk 3 above)
Comment (karlprieb) [reply bats_test/create-drive/drive-name-validation.bats]: [926db0a](https://github.com/ardriveapp/ardrive-cli/pull/247/commits/926db0a540bd62fbd7070aa66f12ff6f385d002e)

File: bats_test/create-drive/drive-name-validation.bats (hunk 2 above)
Comment (karlprieb) [reply bats_test/create-drive/drive-name-validation.bats]: I create a constant for this string [926db0a](https://github.com/ardriveapp/ardrive-cli/pull/247/commits/926db0a540bd62fbd7070aa66f12ff6f385d002e)

**Review by fedellen — APPROVED** [review]

**Review by fedellen — APPROVED** [review]

**Review by matibat — APPROVED** [review]
LGTM

**Pull request merged**